### ISSUE TITLE
Segfault running phpunit

### ISSUE THREAD

**Issue by remicollet** [scoutapp/scout-apm-php-ext#55]
Hi,

Running doctrine/migrations 2.2.0 test suite

```
#0  0x0000000000000000 in  ()
#1  0x00007ffff7fcb440 in scoutapm_default_handler (execute_data=0x7ffff6e249a0, return_value=0x7ffff6e24930)
    at /usr/src/debug/php-pecl-scoutapm-1.0.1-1.fc29.remi.7.2.x86_64/NTS/zend_scoutapm.c:141
#2  0x00007fffe824c575 in pt_execute_core (internal=1, execute_data=0x7ffff6e249a0, return_value=0x7ffff6e24930)
    at /usr/src/debug/php-pecl-trace-1.0.1-0.5.beta.fc29.remi.7.2.x86_64/trace-1.0.1beta/extension/trace.c:1335
#3  0x0000555555850198 in ZEND_DO_FCALL_SPEC_RETVAL_USED_HANDLER () at /usr/src/debug/php-7.2.24-1.fc29.remi.x86_64/Zend/zend_vm_execute.h:1038
#4  0x00005555558a9e2f in execute_ex (ex=0x7ffff6e249a0) at /usr/src/debug/php-7.2.24-1.fc29.remi.x86_64/Zend/zend_vm_execute.h:59832
#5  0x00007fffe824c4cb in pt_execute_core (internal=0, execute_data=0x7ffff6e24890, return_value=0x0)
    at /usr/src/debug/php-pecl-trace-1.0.1-0.5.beta.fc29.remi.7.2.x86_64/trace-1.0.1beta/extension/trace.c:1338
#6  0x000055555585027c in ZEND_DO_FCALL_SPEC_RETVAL_USED_HANDLER () at /usr/src/debug/php-7.2.24-1.fc29.remi.x86_64/Zend/zend_vm_execute.h:1017
#7  0x00005555558a9e2f in execute_ex (ex=0x7ffff6e249a0) at /usr/src/debug/php-7.2.24-1.fc29.remi.x86_64/Zend/zend_vm_execute.h:59832
#8  0x00007fffe824c4cb in pt_execute_core (internal=0, execute_data=0x7ffff6e24600, return_value=0x0)
    at /usr/src/debug/php-pecl-trace-1.0.1-0.5.beta.fc29.remi.7.2.x86_64/trace-1.0.1beta/extension/trace.c:1338
#9  0x000055555585027c in ZEND_DO_FCALL_SPEC_RETVAL_USED_HANDLER () at /usr/src/debug/php-7.2.24-1.fc29.remi.x86_64/Zend/zend_vm_execute.h:1017
#10 0x00005555558a9e2f in execute_ex (ex=0x7ffff6e249a0) at /usr/src/debug/php-7.2.24-1.fc29.remi.x86_64/Zend/zend_vm_execute.h:59832
#11 0x00007fffe824c4cb in pt_execute_core (internal=0, execute_data=0x7ffff6e24560, return_value=0x0)
    at /usr/src/debug/php-pecl-trace-1.0.1-0.5.beta.fc29.remi.7.2.x86_64/trace-1.0.1beta/extension/trace.c:1338
#12 0x000055555585027c in ZEND_DO_FCALL_SPEC_RETVAL_USED_HANDLER () at /usr/src/debug/php-7.2.24-1.fc29.remi.x86_64/Zend/zend_vm_execute.h:1017
#13 0x00005555558a9e2f in execute_ex (ex=0x7ffff6e249a0) at /usr/src/debug/php-7.2.24-1.fc29.remi.x86_64/Zend/zend_vm_execute.h:59832
#14 0x00007fffe824c4cb in pt_execute_core (internal=0, execute_data=0x7ffff6e24450, return_value=0x0)
    at /usr/src/debug/php-pecl-trace-1.0.1-0.5.beta.fc29.remi.7.2.x86_64/trace-1.0.1beta/extension/trace.c:1338
#15 0x000055555585027c in ZEND_DO_FCALL_SPEC_RETVAL_USED_HANDLER () at /usr/src/debug/php-7.2.24-1.fc29.remi.x86_64/Zend/zend_vm_execute.h:1017
#16 0x00005555558a9e2f in execute_ex (ex=0x7ffff6e249a0) at /usr/src/debug/php-7.2.24-1.fc29.remi.x86_64/Zend/zend_vm_execute.h:59832
#17 0x00007fffe824c4cb in pt_execute_core (internal=0, execute_data=0x7ffff6e24280, return_value=0x0)
    at /usr/src/debug/php-pecl-trace-1.0.1-0.5.beta.fc29.remi.7.2.x86_64/trace-1.0.1beta/extension/trace.c:1338
#18 0x000055555585027c in ZEND_DO_FCALL_SPEC_RETVAL_USED_HANDLER () at /usr/src/debug/php-7.2.24-1.fc29.remi.x86_64/Zend/zend_vm_execute.h:1017
#19 0x00005555558a9e2f in execute_ex (ex=0x7ffff6e249a0) at /usr/src/debug/php-7.2.24-1.fc29.remi.x86_64/Zend/zend_vm_execute.h:59832
#20 0x00007fffe824c4cb in pt_execute_core (internal=0, execute_data=0x7ffff6e24190, return_value=0x0)
    at /usr/src/debug/php-pecl-trace-1.0.1-0.5.beta.fc29.remi.7.2.x86_64/trace-1.0.1beta/extension/trace.c:1338
#21 0x000055555585027c in ZEND_DO_FCALL_SPEC_RETVAL_USED_HANDLER () at /usr/src/debug/php-7.2.24-1.fc29.remi.x86_64/Zend/zend_vm_execute.h:1017
#22 0x00005555558a9e2f in execute_ex (ex=0x7ffff6e249a0) at /usr/src/debug/php-7.2.24-1.fc29.remi.x86_64/Zend/zend_vm_execute.h:59832
#23 0x00007fffe824c4cb in pt_execute_core (internal=0, execute_data=0x7ffff6e23ff0, return_value=0x0)
    at /usr/src/debug/php-pecl-trace-1.0.1-0.5.beta.fc29.remi.7.2.x86_64/trace-1.0.1beta/extension/trace.c:1338
#24 0x000055555585027c in ZEND_DO_FCALL_SPEC_RETVAL_USED_HANDLER () at /usr/src/debug/php-7.2.24-1.fc29.remi.x86_64/Zend/zend_vm_execute.h:1017
#25 0x00005555558a9e2f in execute_ex (ex=0x7ffff6e249a0) at /usr/src/debug/php-7.2.24-1.fc29.remi.x86_64/Zend/zend_vm_execute.h:59832
#26 0x00007fffe824c4cb in pt_execute_core (internal=0, execute_data=0x7ffff6e23f70, return_value=0x0)
    at /usr/src/debug/php-pecl-trace-1.0.1-0.5.beta.fc29.remi.7.2.x86_64/trace-1.0.1beta/extension/trace.c:1338
#27 0x000055555584fe14 in ZEND_DO_FCALL_SPEC_RETVAL_UNUSED_HANDLER () at /usr/src/debug/php-7.2.24-1.fc29.remi.x86_64/Zend/zend_vm_execute.h:892
#28 0x00005555558a9e2f in execute_ex (ex=0x7ffff6e249a0) at /usr/src/debug/php-7.2.24-1.fc29.remi.x86_64/Zend/zend_vm_execute.h:59832
#29 0x00007fffe824c4cb in pt_execute_core (internal=0, execute_data=0x7ffff6e23c20, return_value=0x0)
    at /usr/src/debug/php-pecl-trace-1.0.1-0.5.beta.fc29.remi.7.2.x86_64/trace-1.0.1beta/extension/trace.c:1338
#30 0x000055555585027c in ZEND_DO_FCALL_SPEC_RETVAL_USED_HANDLER () at /usr/src/debug/php-7.2.24-1.fc29.remi.x86_64/Zend/zend_vm_execute.h:1017
#31 0x00005555558a9e2f in execute_ex (ex=0x7ffff6e249a0) at /usr/src/debug/php-7.2.24-1.fc29.remi.x86_64/Zend/zend_vm_execute.h:59832
#32 0x00007fffe824c4cb in pt_execute_core (internal=0, execute_data=0x7ffff6e23ba0, return_value=0x0)
    at /usr/src/debug/php-pecl-trace-1.0.1-0.5.beta.fc29.remi.7.2.x86_64/trace-1.0.1beta/extension/trace.c:1338
#33 0x000055555585027c in ZEND_DO_FCALL_SPEC_RETVAL_USED_HANDLER () at /usr/src/debug/php-7.2.24-1.fc29.remi.x86_64/Zend/zend_vm_execute.h:1017
#34 0x00005555558a9e2f in execute_ex (ex=0x7ffff6e249a0) at /usr/src/debug/php-7.2.24-1.fc29.remi.x86_64/Zend/zend_vm_execute.h:59832
#35 0x00007fffe824c4cb in pt_execute_core (internal=0, execute_data=0x7ffff6e23a00, return_value=0x0)
    at /usr/src/debug/php-pecl-trace-1.0.1-0.5.beta.fc29.remi.7.2.x86_64/trace-1.0.1beta/extension/trace.c:1338
#36 0x000055555585027c in ZEND_DO_FCALL_SPEC_RETVAL_USED_HANDLER () at /usr/src/debug/php-7.2.24-1.fc29.remi.x86_64/Zend/zend_vm_execute.h:1017
#37 0x00005555558a9e2f in execute_ex (ex=0x7ffff6e249a0) at /usr/src/debug/php-7.2.24-1.fc29.remi.x86_64/Zend/zend_vm_execute.h:59832
#38 0x00007fffe824c4cb in pt_execute_core (internal=0, execute_data=0x7ffff6e23560, return_value=0x0)
    at /usr/src/debug/php-pecl-trace-1.0.1-0.5.beta.fc29.remi.7.2.x86_64/trace-1.0.1beta/extension/trace.c:1338
#39 0x000055555585027c in ZEND_DO_FCALL_SPEC_RETVAL_USED_HANDLER () at /usr/src/debug/php-7.2.24-1.fc29.remi.x86_64/Zend/zend_vm_execute.h:1017
#40 0x00005555558a9e2f in execute_ex (ex=0x7ffff6e249a0) at /usr/src/debug/php-7.2.24-1.fc29.remi.x86_64/Zend/zend_vm_execute.h:59832
#41 0x00007fffe824c4cb in pt_execute_core (internal=0, execute_data=0x7ffff6e22de0, return_value=0x0)
    at /usr/src/debug/php-pecl-trace-1.0.1-0.5.beta.fc29.remi.7.2.x86_64/trace-1.0.1beta/extension/trace.c:1338
#42 0x000055555584fe14 in ZEND_DO_FCALL_SPEC_RETVAL_UNUSED_HANDLER () at /usr/src/debug/php-7.2.24-1.fc29.remi.x86_64/Zend/zend_vm_execute.h:892
#43 0x00005555558a9e2f in execute_ex (ex=0x7ffff6e249a0) at /usr/src/debug/php-7.2.24-1.fc29.remi.x86_64/Zend/zend_vm_execute.h:59832
#44 0x00007fffe824c4cb in pt_execute_core (internal=0, execute_data=0x7ffff6e21d20, return_value=0x0)
    at /usr/src/debug/php-pecl-trace-1.0.1-0.5.beta.fc29.remi.7.2.x86_64/trace-1.0.1beta/extension/trace.c:1338
#45 0x000055555584fe14 in ZEND_DO_FCALL_SPEC_RETVAL_UNUSED_HANDLER () at /usr/src/debug/php-7.2.24-1.fc29.remi.x86_64/Zend/zend_vm_execute.h:892
#46 0x00005555558a9e2f in execute_ex (ex=0x7ffff6e249a0) at /usr/src/debug/php-7.2.24-1.fc29.remi.x86_64/Zend/zend_vm_execute.h:59832
#47 0x00007fffe824c4cb in pt_execute_core (internal=0, execute_data=0x7ffff6e21200, return_value=0x0)
    at /usr/src/debug/php-pecl-trace-1.0.1-0.5.beta.fc29.remi.7.2.x86_64/trace-1.0.1beta/extension/trace.c:1338
#48 0x000055555584fe14 in ZEND_DO_FCALL_SPEC_RETVAL_UNUSED_HANDLER () at /usr/src/debug/php-7.2.24-1.fc29.remi.x86_64/Zend/zend_vm_execute.h:892
#49 0x00005555558a9e2f in execute_ex (ex=0x7ffff6e249a0) at /usr/src/debug/php-7.2.24-1.fc29.remi.x86_64/Zend/zend_vm_execute.h:59832
#50 0x00007fffe824c4cb in pt_execute_core (internal=0, execute_data=0x7ffff6e20b20, return_value=0x0)
    at /usr/src/debug/php-pecl-trace-1.0.1-0.5.beta.fc29.remi.7.2.x86_64/trace-1.0.1beta/extension/trace.c:1338
#51 0x000055555584fe14 in ZEND_DO_FCALL_SPEC_RETVAL_UNUSED_HANDLER () at /usr/src/debug/php-7.2.24-1.fc29.remi.x86_64/Zend/zend_vm_execute.h:892
#52 0x00005555558a9e2f in execute_ex (ex=0x7ffff6e249a0) at /usr/src/debug/php-7.2.24-1.fc29.remi.x86_64/Zend/zend_vm_execute.h:59832
#53 0x00007fffe824c4cb in pt_execute_core (internal=0, execute_data=0x7ffff6e20440, return_value=0x0)
    at /usr/src/debug/php-pecl-trace-1.0.1-0.5.beta.fc29.remi.7.2.x86_64/trace-1.0.1beta/extension/trace.c:1338
#54 0x000055555584fe14 in ZEND_DO_FCALL_SPEC_RETVAL_UNUSED_HANDLER () at /usr/src/debug/php-7.2.24-1.fc29.remi.x86_64/Zend/zend_vm_execute.h:892
#55 0x00005555558a9e2f in execute_ex (ex=0x7ffff6e249a0) at /usr/src/debug/php-7.2.24-1.fc29.remi.x86_64/Zend/zend_vm_execute.h:59832
#56 0x00007fffe824c4cb in pt_execute_core (internal=0, execute_data=0x7ffff6e1fd60, return_value=0x0)
    at /usr/src/debug/php-pecl-trace-1.0.1-0.5.beta.fc29.remi.7.2.x86_64/trace-1.0.1beta/extension/trace.c:1338
#57 0x000055555584fe14 in ZEND_DO_FCALL_SPEC_RETVAL_UNUSED_HANDLER () at /usr/src/debug/php-7.2.24-1.fc29.remi.x86_64/Zend/zend_vm_execute.h:892
#58 0x00005555558a9e2f in execute_ex (ex=0x7ffff6e249a0) at /usr/src/debug/php-7.2.24-1.fc29.remi.x86_64/Zend/zend_vm_execute.h:59832
#59 0x00007fffe824c4cb in pt_execute_core (internal=0, execute_data=0x7ffff6e1d740, return_value=0x0)
    at /usr/src/debug/php-pecl-trace-1.0.1-0.5.beta.fc29.remi.7.2.x86_64/trace-1.0.1beta/extension/trace.c:1338
#60 0x000055555585027c in ZEND_DO_FCALL_SPEC_RETVAL_USED_HANDLER () at /usr/src/debug/php-7.2.24-1.fc29.remi.x86_64/Zend/zend_vm_execute.h:1017
#61 0x00005555558a9e2f in execute_ex (ex=0x7ffff6e249a0) at /usr/src/debug/php-7.2.24-1.fc29.remi.x86_64/Zend/zend_vm_execute.h:59832
#62 0x00007fffe824c4cb in pt_execute_core (internal=0, execute_data=0x7ffff6e1d350, return_value=0x0)
    at /usr/src/debug/php-pecl-trace-1.0.1-0.5.beta.fc29.remi.7.2.x86_64/trace-1.0.1beta/extension/trace.c:1338
#63 0x000055555585027c in ZEND_DO_FCALL_SPEC_RETVAL_USED_HANDLER () at /usr/src/debug/php-7.2.24-1.fc29.remi.x86_64/Zend/zend_vm_execute.h:1017
#64 0x00005555558a9e2f in execute_ex (ex=0x7ffff6e249a0) at /usr/src/debug/php-7.2.24-1.fc29.remi.x86_64/Zend/zend_vm_execute.h:59832
#65 0x00007fffe824c4cb in pt_execute_core (internal=0, execute_data=0x7ffff6e1d280, return_value=0x0)
    at /usr/src/debug/php-pecl-trace-1.0.1-0.5.beta.fc29.remi.7.2.x86_64/trace-1.0.1beta/extension/trace.c:1338
#66 0x000055555584fe14 in ZEND_DO_FCALL_SPEC_RETVAL_UNUSED_HANDLER () at /usr/src/debug/php-7.2.24-1.fc29.remi.x86_64/Zend/zend_vm_execute.h:892
#67 0x00005555558a9e2f in execute_ex (ex=0x7ffff6e249a0) at /usr/src/debug/php-7.2.24-1.fc29.remi.x86_64/Zend/zend_vm_execute.h:59832
#68 0x00007fffe824c4cb in pt_execute_core (internal=0, execute_data=0x7ffff6e1d030, return_value=0x0)
    at /usr/src/debug/php-pecl-trace-1.0.1-0.5.beta.fc29.remi.7.2.x86_64/trace-1.0.1beta/extension/trace.c:1338
#69 0x00005555558b3076 in zend_execute (op_array=0x7ffff6e78a80, op_array@entry=0x7fffcf8cb080, return_value=return_value@entry=0x0)
    at /usr/src/debug/php-7.2.24-1.fc29.remi.x86_64/Zend/zend_vm_execute.h:63780
#70 0x000055555580cf8b in zend_execute_scripts (type=type@entry=8, retval=retval@entry=0x0, file_count=file_count@entry=3)
    at /usr/src/debug/php-7.2.24-1.fc29.remi.x86_64/Zend/zend.c:1498
#71 0x00005555557a8560 in php_execute_script (primary_file=<optimized out>) at /usr/src/debug/php-7.2.24-1.fc29.remi.x86_64/main/main.c:2599
#72 0x00005555558b5650 in do_cli (argc=5, argv=0x555555a4c120) at /usr/src/debug/php-7.2.24-1.fc29.remi.x86_64/sapi/cli/php_cli.c:1011
#73 0x0000555555661c72 in main (argc=5, argv=0x555555a4c120) at /usr/src/debug/php-7.2.24-1.fc29.remi.x86_64/sapi/cli/php_cli.c:1403

```

**Comment by cschneid** [comment]
Thank you for the report, and sorry for the issue. Which version of the extension are you on? We fixed a potential segfault in 1.0.1 if you happen to be on 1.0.0 still.

We'll take a look at this, thank you!

**Comment by cschneid** [comment]
Oh, I see 1.0.1 in the trace. We'll take a look
